### PR TITLE
feat: use indexer endpoint for affiliate stats

### PR DESCRIPTION
**Pull request by tinaszheng**
This PR switches over to the indexer endpoint to retrieve affiliate stats. Note that there are different field names.

This also simplifies the affiliate leaderboard by removing pagination (and always just showing top 100).

POV someone on the leaderboard: 
image

POV no affiliate stats yet:
image

### CODE REVIEW

File: src/hooks/useAffiliatesInfo.ts [hunk 1]
```diff
@@ -114,37 +77,24 @@ export const useAffiliatesInfo = (dydxAddress?: string) => {
...
     } catch (error) {
-      log('useAffiliatesInfo/fetchLastUpdated', error, { endpoint });
+      log('useAffiliatesInfo/fetchAccountStats', error, { endpoint });
       throw error;
     }
   };
 
   const affiliateMetadataQuery = useQuery({
-    queryKey: ['affiliateMetadata', dydxAddress, compositeClient],
+    queryKey: ['affiliateMetadata', dydxAddress],
```
Comment (tinaszheng) [inline src/hooks/useAffiliatesInfo.ts:89]: compositeClient doesnt need to be a queryKey here

File: src/hooks/useAffiliatesInfo.ts [hunk 2]
```diff
@@ -114,37 +77,24 @@ export const useAffiliatesInfo = (dydxAddress?: string) => {
...
     queryFn: fetchAffiliateMetadata,
     enabled: Boolean(compositeClient && dydxAddress && affiliatesBaseUrl),
   });
 
-  const programStatsQuery = useQuery({
```
Comment (tinaszheng) [inline src/hooks/useAffiliatesInfo.ts:131]: removing this because its unused now

File: src/views/Affiliates/cards/AffiliateStatsCard.tsx [hunk 3]
```diff
@@ -33,10 +33,10 @@ const MobileView = ({
 
   return (
     <div tw="flex flex-wrap items-center justify-between">
-      <MobileStatsHeader tw="flex w-full justify-between gap-1 divide-y p-1">
+      <MobileStatsHeader tw="flex w-full gap-1 divide-y p-1">
```
Comment (tinaszheng) [inline src/views/Affiliates/cards/AffiliateStatsCard.tsx:36]: now that theres only two cells it looks better with flex-1 on both

File: src/hooks/useAffiliatesInfo.ts [hunk 4]
```diff
@@ -63,47 +63,10 @@ export const useAffiliatesInfo = (dydxAddress?: string) => {
...
-      return data;
-    } catch (error) {
-      log('useAffiliatesInfo/fetchAccountStats', error, { endpoint });
-      throw error;
-    }
-  };
-
-  const fetchLastUpdated = async () => {
-    const endpoint = `${affiliatesBaseUrl}/v1/last-updated`;
+    const endpoint = `${compositeClient.indexerClient.config.restEndpoint}/v4/affiliates/snapshot?addressFilter=${encodeURIComponent(dydxAddress)}`;
```
Comment (tyleroooo) [inline src/hooks/useAffiliatesInfo.ts:69]: these really should live in v4-clients at this point but seems okay for now

**Review by tyleroooo — APPROVED** [review]

**Review by jaredvu — APPROVED** [review]

**Pull request merged**